packages/node/src/events/protocol-message/controller.ts

```diff
@@ -25,4 +30,18 @@ export default async function protocolMessageEventController(
     async stateChannel =>
       await requestHandler.store.saveStateChannel(stateChannel)
   );
+
+  // TODO: Follow this pattern for all machine related events
+  if (nodeMsg.data.protocol === Protocol.UninstallVirtualApp) {
+    const uninstallMsg: UninstallMessage = {
+      from: requestHandler.publicIdentifier,
+      type: NODE_EVENTS.UNINSTALL_VIRTUAL,
+      data: {
+        appInstanceId: (nodeMsg.data.params as UninstallVirtualAppParams)
+          .targetAppIdentityHash
+      }
+    };
+
+    requestHandler.outgoing.emit(uninstallMsg.type, uninstallMsg);
```

Comment:
So this event is being sent out twice?

It's also being sent out outside of being handled here, right?

Reply:
Why is this being sent out to begin with?

Reply:
This is the `UNINSTALL_VIRTUAL` event that the UI needs to listen to in order to confirm the app has been uninstalled.

Reply:
Also, just sent out once.